### PR TITLE
When multiple custom resource handlers are subscribed, the first wins

### PR DESCRIPTION
**Pull request by joaompneves**

### CODE REVIEW

File: ReactViewControl/ReactViewRender.cs [hunk 1]
```diff
@@ -394,26 +402,38 @@ private enum LoadStatus {
...
-                ExternalResourceRequested?.Invoke(resourceHandler);
+                case ResourceUrl.EmbeddedScheme:
+                    // webview already started BeginAsyncResponse
+                    EmbeddedResourceRequested?.Invoke(resourceHandler);
+                    break;
+
+                case "http":
+                case "https":
+                    ExternalResourceRequested?.Invoke(resourceHandler);
+                    break;
```
Comment (alvesmiguel1) [inline ReactViewControl/ReactViewRender.cs]: default break not needed?

File: ReactViewControl/ReactViewRender.cs [hunk 2]
```diff
@@ -394,26 +402,38 @@ private enum LoadStatus {
...
-                            if (Uri.TryCreate(url, UriKind.Absolute, out var uri)) {
-                                var path = uri.AbsolutePath;
-                                extension = Path.GetExtension(path).TrimStart('.');
+            var url = resourceHandler.Url;
+            var scheme = url.Substring(0, Math.Max(0, url.IndexOf(Uri.SchemeDelimiter)));
+
+            switch (scheme.ToLowerInvariant()) {
+                case ResourceUrl.CustomScheme:
+                    var customResourceRequestedHandlers = CustomResourceRequested?.GetInvocationList().ToArray();
+                    if (customResourceRequestedHandlers?.Any() == true) {
```
Comment (alvesmiguel1) [inline ReactViewControl/ReactViewRender.cs]: Or (customResourceRequestedHandlers?.Any() ?? false) 😄

File: ReactViewControl/ReactViewRender.cs [hunk 3]
```diff
@@ -394,26 +402,38 @@ private enum LoadStatus {
...
-                        var url = resourceHandler.Url;
-                        var response = customResourceRequested(url);
-
-                        if (response != null) {
-                            string extension = null;
-                            if (Uri.TryCreate(url, UriKind.Absolute, out var uri)) {
-                                var path = uri.AbsolutePath;
-                                extension = Path.GetExtension(path).TrimStart('.');
+            var url = resourceHandler.Url;
+            var scheme = url.Substring(0, Math.Max(0, url.IndexOf(Uri.SchemeDelimiter)));
```
Comment (alvesmiguel1) [inline ReactViewControl/ReactViewRender.cs]: You might use `Uri.TryCreate(url, UriKind.Absolute, out var uri)` here, which you already use below, and then use uri.Scheme

File: WebViewControl/WebView.CefResourceHandlerFactory.cs [hunk 4]
```diff
@@ -55,32 +55,30 @@ protected class CefResourceHandlerFactory : IResourceHandlerFactory, ISchemeHand
...
-                                OwnerWebView.ExecuteWithAsyncErrorHandling(() => throw new InvalidOperationException("Resource not found: " + url));
-                            }
-                        });
-
-                        return;
-                    }
+                if (Uri.TryCreate(resourceHandler.Url, UriKind.Absolute, out var url) && url.Scheme == ResourceUrl.EmbeddedScheme) {
+                    resourceHandler.BeginAsyncResponse(() => {
+                        var urlWithoutQuery = new UriBuilder(url);
+                        if (url.Query != "") {
```
Comment (alvesmiguel1) [inline WebViewControl/WebView.CefResourceHandlerFactory.cs]: string.Empty

Comment (alvesmiguel1) [reply WebViewControl/WebView.CefResourceHandlerFactory.cs]: Same below

File: ReactViewControl/ReactViewRender.cs (hunk 1 above)
Comment (joaompneves) [reply ReactViewControl/ReactViewRender.cs]: nope

File: ReactViewControl/ReactViewRender.cs (hunk 2 above)
Comment (joaompneves) [reply ReactViewControl/ReactViewRender.cs]: when using this syntax pattern I prefer the == true ... when using the "?? false" I have to thing twice (I have limitations)

**Pull request merged**